src/private/EventMetricType.js

```diff
@@ -10,7 +10,7 @@ class EventMetricType {
     this.name = name;
   }
 
-  async record(extra) {
```

Comment:
`async` is also not supported in QML envs, we were not using the asyncness of oyr functions anyways, namely we were not `await`ing anything, so there is no harm in removing these annotations.

In case we want to do `async/await` stuff we can always have a function that returns a Promise and do the thing when the Promise is resolved/reject the old fashioned way (see: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Using_promises)